taraxa/state/dpos_2.0/precompiled/delegations.go

```diff
@@ -0,0 +1,123 @@
+package dpos_2
+
+import (
+	"math/big"
+
+	"github.com/Taraxa-project/taraxa-evm/common"
+	"github.com/Taraxa-project/taraxa-evm/core/types"
+	"github.com/Taraxa-project/taraxa-evm/rlp"
+)
+
+type Delegation struct {
+	// Num of delegated tokens == delegator's stake
+	Stake *big.Int
+
+	// Block number related to rewards
+	LastUpdated types.BlockNum
+}
+
+// Delegations type groups together all functionality related to creating/deleting/modifying/etc... delegations
+// as such info is stored under multiple independent storage keys, it is important that caller does not need to
+// think about all implementation details, but just calls functions on Delegations type
+type Delegations struct {
+	storage StorageWrapper
+	// <delegator addres -> list of validators> as each delegator can delegate to multiple validators
+	delegators_validators map[common.Address]*IterableMap
+
+	delegations_field                  []byte
+	delegators_validators_field_prefix []byte
+}
+
+func (self *Delegations) Init(stor StorageWrapper, prefix []byte) {
+	self.storage = stor
+
+	// Init Delegations storage fields keys - relative to the prefix
+	self.delegations_field = append(prefix, []byte{0}...)
+	self.delegators_validators_field_prefix = append(prefix, []byte{1}...)
+}
+
+// Checks if delegation exists
+func (self *Delegations) DelegationExists(delegator_address *common.Address, validator_address *common.Address) bool {
```

Comment:
I do not understand why we are not also returning the key? all those functions could be faster, if we calculate key just once, or let's have 2 versions